Tests/RudderStackAnalyticsTests/SourceConfigModuleTests/SourceConfigProviderTests.swift

```diff
@@ -0,0 +1,371 @@
+//
+//  SourceConfigProviderTests.swift
+//  RudderStackAnalyticsTests
+//
+//  Created by Satheesh Kannan on 25/10/25.
+//
+
+import Foundation
+import Testing
+import Combine
+@testable import RudderStackAnalytics
+
+@Suite("SourceConfigProvider Tests")
+class SourceConfigProviderTests {
+    
+    var mockStorage: MockStorage
+    var analytics: Analytics
+    var provider: SourceConfigProvider
+    
+    init() {
+        self.mockStorage = MockStorage()
+        self.analytics = SwiftTestMockProvider.createMockAnalytics(storage: mockStorage)
+        self.provider = MockSourceConfigProvider(analytics: analytics)
```

Comment:
We should try to create an instance of the class under test directly within the test class. Try not to use Mock word.

I know in `MockSourceConfigProvider` we are creating an actual instance but let's move that logic over here. The word `Mock` is confusing.

```suggestion
        self.provider = ExponentialBackoffPolicy(minDelayInMillis: 50) // 0.05 secs
```

Reply:
Also, do we really need the 50 milli-sec delay? Is it possible to keep it 0sec, if it doesn't cause any side-effect.

Reply:
1. In the `MockSourceConfigProvider` implementation, the `ExponentialBackoffPolicy` is configured through the `provideBackoffPolicy()` method. Would you prefer to create a dedicated mock for `MockSourceConfigProvider`, or initialize it directly within each test case, or rename the `MockSourceConfigProvider` class to more meaningful name?

2. We can set the `minDelayInMillis` value to `0`, which means it will always return `0` as the `delay`.
However, this also means the delay won’t increase incrementally between retries. Is that fine?

Reply:
1. Let me clarify this first: I'm not proposing to modify the `MockSourceConfigProvider`. You can define and use it in other places. That's fine. But in the `SourceConfigProviderTest` let's create an actual instance instead of the Mock. Refer to [this](https://github.com/rudderlabs/rudder-sdk-kotlin/blob/9bdbe82f854ff5c95a908c9ef59ed805bd9e31a1/core/src/test/kotlin/com/rudderstack/sdk/kotlin/core/SourceConfigManagerTest.kt#L65) `Kotlin` example for reference.
2. If there’s an issue with verifying our business logic, then it’s not fine. My only concern with setting the `delay` to anything other than 0 is that it will increase the test duration. Is it possible to mock the `delay` so the tests execute without any actual delay? If tests are already executing without the actual `delay` then there's no need to modify the test.